keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -46,80 +49,82 @@ class ResourceActuator(
   suspend fun <T : ResourceSpec> checkResource(resource: Resource<T>) {
     val id = resource.id
     val plugin = handlers.supporting(resource.apiVersion, resource.kind)
-
-    // todo eb: extract "application veto" as its own concept, and check before generating the diff.
-
-    if (plugin.actuationInProgress(resource)) {
-      log.debug("Actuation for resource {} is already running, skipping checks", id)
-      publisher.publishEvent(ResourceCheckSkipped(resource.apiVersion, resource.kind, id, "ActuationInProgress"))
-      return
-    }
-
-    try {
-      val (desired, current) = plugin.resolve(resource)
-      val diff = ResourceDiff(desired, current)
-      if (diff.hasChanges()) {
-        diffFingerprintRepository.store(id, diff)
+    MDC.put("X-SPINNAKER-RESOURCE-ID", id.toString())
```

Comment:
I think is useful overall. We could probably hide or instrument this in some way where it's not as in-your-face, but maybe is a separate method that wraps the `checkResource(*)` logic?

Reply:
Yeah, this was just a quick-and-dirty PoC. I was thinking maybe an annotation might be good?